forum/qa-plugin/q2apro-on-site-notifications/q2apro-onsitenotifications-page.php

```diff
@@ -116,8 +116,12 @@ public function process_request($request)
 
         foreach ($events as $postid_string => $event) {
             // $postid_string, e.g. 32_1 (32 is postid, 1 is global event count)
-            $type = $event['event'];
 
+            if (array_key_exists('handle', $event) && $event['handle'] === null) {
```

Comment:
A co się stanie, jeśli nie użyje się `array_key_exists(..)`? Jeśli w tablicy `$event` nie ma klucza 'handle', to nie można odczytać property, czy zwróci wartość podobną do `undefined` lub `null`?

Reply:
jak przy każdej próbie odwołania się do nieistniejącego elementu tablicy - rzuci warning

Reply:
A nie lepiej `isset()`?
https://stackoverflow.com/questions/3210935/whats-the-difference-between-isset-and-array-key-exists

Reply:
Nie lepiej, ponieważ ten warunek powinien się spełnić tylko gdy element `handle` w tablicy `$event` istnieje i jest `null`. Zastosowanie `if (!isset($event['handle']))` spowodowałoby spełnienie warunku zarówno gdy element byłby nullem (tak jak potrzebuję), ale także gdy elementu wcale nie ma (a tu już się spełnić nie może, bo z tego co podałem są w tej tablicy elementy bez tego klucza, które są prawidłowymi powiadomieniami).